src/general/AutoModLoader.cs

```diff
@@ -0,0 +1,65 @@
+using System.IO;
+using System.Reflection;
+using Godot;
+using Newtonsoft.Json;
+using Directory = System.IO.Directory;
+using File = System.IO.File;
+
+public class AutoModLoader : Node
+{
+    private AutoModLoader()
+    {
+        DirectoryInfo modFolder = Directory.CreateDirectory($"{Directory.GetCurrentDirectory()}\\mods");
```

Comment:
Mods should really be in the Godot "user://" directory as the game could be installed in a read only (or administrator writable) folder. So mods need to be elsewhere than in the game folder.